app_test.go

```diff
@@ -0,0 +1,250 @@
+package flamingo_test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"flamingo.me/dingo"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"flamingo.me/flamingo/v3"
+	"flamingo.me/flamingo/v3/framework/cmd"
+	framework "flamingo.me/flamingo/v3/framework/flamingo"
+)
+
+type NotifyFunc func(ctx context.Context, event framework.Event)
+
+func (nf NotifyFunc) Notify(ctx context.Context, event framework.Event) {
+	nf(ctx, event)
+}
+
+func TestCmdEventsTriggeredProperly(t *testing.T) { //nolint:paralleltest // due to dingo.Singleton
+	assertStartupAndShutdownOnce := func(startupEventCount, shutdownEventCount int32) {
```

Comment:
this function should receive the `testing.T` from the subtest (as first arg and be `t.Helper()`)